### PR TITLE
[Task 1872013] Support default value(region, runtime, pricing tier) for webapp and functions

### PR DESCRIPTION
**Pull request by andxu**
[AB#1872013](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1872013)

### CODE REVIEW

File: azure-functions-gradle-plugin/src/main/java/com/microsoft/azure/plugin/functions/gradle/handler/DeployHandler.java [hunk 1]
```diff
@@ -275,12 +276,36 @@ protected void validateParameters() {
...
+        boolean createFunctionApp = app.exists();
+        final AppServiceConfig defaultConfig = createFunctionApp ? buildDefaultConfig(functionConfig.subscriptionId(),
+            functionConfig.resourceGroup(), functionConfig.appName()) : fromAppService(app, app.plan());
+        mergeAppServiceConfig(functionConfig, defaultConfig);
+        if (!createFunctionApp && !functionConfig.disableAppInsights()) {
+            // fill ai key from existing app settings
+            functionConfig.appInsightsKey(app.entity().getAppSettings().get(CreateOrUpdateFunctionAppTask.APPINSIGHTS_INSTRUMENTATION_KEY));
+        }
+        if (!createFunctionApp) {
+            return createFunctionApp(functionConfig);
```
Comment (Flanker32) [inline azure-functions-gradle-plugin/src/main/java/com/microsoft/azure/plugin/functions/gradle/handler/DeployHandler.java:306]: may call new CreateOrUpdateFunctionAppTask(functionConfig).execute() directly here

**Review by Flanker32 — DISMISSED** [review]
LGTM

File: azure-functions-gradle-plugin/src/main/java/com/microsoft/azure/plugin/functions/gradle/handler/DeployHandler.java [hunk 2]
```diff
@@ -166,6 +154,19 @@ public void execute() throws AzureExecutionException {
...
+    private RuntimeConfig getRuntimeConfig() {
+        final GradleRuntimeConfig runtime = ctx.getRuntime();
+        if (runtime == null) {
+            return null;
+        }
+        final OperatingSystem os = Optional.ofNullable(runtime.os()).map(OperatingSystem::fromString).orElse(null);
+        final JavaVersion javaVersion = Optional.ofNullable(runtime.javaVersion()).map(JavaVersion::fromString).orElse(null);
+        final RuntimeConfig result = new RuntimeConfig().os(os).javaVersion(javaVersion).webContainer(WebContainer.JAVA_OFF)
+            .image(runtime.image()).registryUrl(runtime.registryUrl());
+        result.username(runtime.username()).password(runtime.password());
```
Comment (Flanker32) [inline azure-functions-gradle-plugin/src/main/java/com/microsoft/azure/plugin/functions/gradle/handler/DeployHandler.java:166]: may add the chain to above statement

File: azure-functions-gradle-plugin/src/main/java/com/microsoft/azure/plugin/functions/gradle/handler/DeployHandler.java [hunk 3]
```diff
@@ -166,6 +154,19 @@ public void execute() throws AzureExecutionException {
         listHTTPTriggerUrls(app);
     }
 
+    private RuntimeConfig getRuntimeConfig() {
+        final GradleRuntimeConfig runtime = ctx.getRuntime();
```
Comment (Flanker32) [inline azure-functions-gradle-plugin/src/main/java/com/microsoft/azure/plugin/functions/gradle/handler/DeployHandler.java:158]: Do we still need `GradleRuntimeConfig` can we get config for task from ctx directly?

**Review by Flanker32 — APPROVED** [review]
LGTM

File: azure-functions-gradle-plugin/src/main/java/com/microsoft/azure/plugin/functions/gradle/handler/DeployHandler.java (hunk 3 above)
Comment (andxu) [reply azure-functions-gradle-plugin/src/main/java/com/microsoft/azure/plugin/functions/gradle/handler/DeployHandler.java:158]: all the properties in GradleRuntimeConfig is string

```java
public class GradleRuntimeConfig {
    private String os;
    private String javaVersion;
    private String webContainer;
    private String image;
    private String registryUrl;
    private String username;
    private String password;
    private String startUpCommand;
}

```

File: azure-functions-gradle-plugin/src/main/java/com/microsoft/azure/plugin/functions/gradle/handler/DeployHandler.java (hunk 2 above)
Comment (andxu) [reply azure-functions-gradle-plugin/src/main/java/com/microsoft/azure/plugin/functions/gradle/handler/DeployHandler.java:166]: fixed in another PR

**Pull request merged**